samza-core/src/main/java/org/apache/samza/operators/impl/OperatorImplGraph.java

```diff
@@ -129,7 +129,7 @@ public void close() {
    */
   OperatorImpl createAndRegisterOperatorImpl(OperatorSpec prevOperatorSpec, OperatorSpec operatorSpec,
       Config config, TaskContext context) {
-    if (!operatorImpls.containsKey(operatorSpec) || operatorSpec instanceof JoinOperatorSpec) {
+    if (!operatorImpls.containsKey(operatorSpec.getOpName()) || operatorSpec instanceof JoinOperatorSpec) {
```

Comment:
@nickpan47 There are 2 PartialJoinOperatorImpl instances corresponding to the same JoinOperatorSpec, which we need to distinguish between. Hence using their names (E.g. join-2-L and join-2-R) as the map keys.